.github/workflows/main.yaml

```diff
@@ -0,0 +1,307 @@
+            SOURCE_IMAGE_REGISTRY: "artifacts.developer.gov.bc.ca/docker-remote/"
+            SOURCE_IMAGE_NAME: caddy
+            SOURCE_IMAGE_TAG: latest
+            REGISTRY_USERNAME_SECRET_NAME: ARTIFACTORY_USERNAME 
+            REGISTRY_PASSWORD_SECRET_NAME: ARTIFACTORY_PASSWORD 
+          - service: aries-endorser-api
+            GIT_REPO_URL: hyperledger/aries-endorser-service
+            GIT_REF: ""
+            DOCKER_FILE_PATH: Dockerfile.endorser # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: endorser # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            SOURCE_IMAGE_REGISTRY: artifacts.developer.gov.bc.ca/docker-remote/
+            SOURCE_IMAGE_NAME: python
+            SOURCE_IMAGE_TAG: 3.10-slim-buster
+            REGISTRY_USERNAME_SECRET_NAME: ARTIFACTORY_USERNAME 
+            REGISTRY_PASSWORD_SECRET_NAME: ARTIFACTORY_PASSWORD 
+    outputs:
+      aries-endorser-agent_digest: ${{ steps.digest.outputs.aries-endorser-agent_digest }} 
+      aries-endorser-backup_digest: ${{ steps.digest.outputs.aries-endorser-backup_digest }} 
+      aries-endorser-api_digest: ${{ steps.digest.outputs.aries-endorser-api_digest }}
+      aries-endorser-proxy_digest: ${{ steps.digest.outputs.aries-endorser-proxy_digest }}
+      aries-endorser-db_digest: ${{ steps.digests.outputs.aries-endorser-db_digest }}  
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with: 
+        repository: ${{ matrix.GIT_REPO_URL }}
+        ref: ${{ matrix.GIT_REF }}
+#        ref: github.ref
+
+        
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to image registry
+      if: matrix.SOURCE_IMAGE_REGISTRY  != '' && contains(fromJSON('["aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ matrix.SOURCE_IMAGE_REGISTRY }}
+        username: ${{ secrets[matrix.REGISTRY_USERNAME_SECRET_NAME]}}
+        password: ${{ secrets[matrix.REGISTRY_PASSWORD_SECRET_NAME]}}
+     
+    - name: Create Docker file
+      if: contains(fromJSON('["aries-endorser-proxy"]'), matrix.service)
+      run: |
+        BASE_IMAGE="${SOURCE_IMAGE_REGISTRY}${SOURCE_IMAGE_NAME}:${SOURCE_IMAGE_TAG}"
+        echo "$BASE_IMAGE"
+        mkdir ${context} && cd ${context} 
+        echo "FROM ${BASE_IMAGE}" > Dockerfile
+        echo "RUN chown 1001:root /usr/bin/caddy" >> Dockerfile
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        SOURCE_IMAGE_REGISTRY: ${{ matrix.SOURCE_IMAGE_REGISTRY }}
+        SOURCE_IMAGE_NAME: ${{ matrix.SOURCE_IMAGE_NAME }}
+        SOURCE_IMAGE_TAG: ${{ matrix.SOURCE_IMAGE_TAG }}
```

Comment:
Use the workflow variables directly in the script rather than assigning them to environment variables, which then get used in the script.